File: Android App - Shredsheets 2/app/src/main/java/revert/shredsheets/models/scales/Diminished.java
package revert.shredsheets.models.scales;

public class Diminished extends Scale {

    @Override
    public int[] getScaleIntervals() {
        return new int[]{2, 1, 2, 1, 2, 1, 2, 1};
    }

    @Override
    public String getName() {
        return "Diminished";
    }

    @Override
    public String[] getModeNames() {
        return new String[]{"Whole Half", "Half Whole"};
    }

    @Override
    protected boolean[][] getModalHighlighting() {
        return new boolean[][]{
                {true, false, true, false, true, false, true, false},
                {true, true, false, true, false, true, false, true},
        };
    }

    @Override
    public int[][] getDegrees() {
        return new int[][]
                {
                        {1, 2, 3, 4, 5, 5, 6, 7},
                        {1, 2, 3, 3, 4, 5, 6, 7},
                };
    }

    @Override
    public String[] getChords() {
        return new String[]{"m7o", "m7o", "m7o", "m7o", "m7o", "m7o", "m7o", "m7o"};
    }
}

File: Shredsheets 3/core/src/com/bm/shredsheets/models/scales/Diminished.kt
package com.bm.shredsheets.models.scales

class Diminished : Scale() {
    override val scaleIntervals: IntArray
        get() = intArrayOf(2, 1, 2, 1, 2, 1, 2, 1)

    override val name: String
        get() = "Diminished"

    override fun getModeNames(): Array<String> {
        return arrayOf("Whole Half", "Half Whole")
    }

    override val getEnum: Scales
        get() = Scales.Diminished

    override val modalHighlighting: Array<BooleanArray>
        protected get() = arrayOf(
            booleanArrayOf(true, false, true, false, true, false, true, false),
            booleanArrayOf(true, true, false, true, false, true, false, true))

    override val degrees: Array<IntArray>
        get() = arrayOf(
            intArrayOf(1, 2, 3, 4, 5, 5, 6, 7),
            intArrayOf(1, 2, 3, 3, 4, 5, 6, 7))

    override val chords: Array<String>
        get() = arrayOf("m7o", "m7o", "m7o", "m7o", "m7o", "m7o", "m7o", "m7o")
}